### PR TITLE
Add server_networks metadata to the inventory (v19.6.1)

### PR DESCRIPTION
**Pull request by brtknr**
This brings the role on par with os-container-infra.

Use case: when the network is not sometimes correctly configured when the instance boots up, having the network IP metadata in the inventory allows us to  configure, e.g. Infiniband interface.

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -34,12 +34,32 @@
...
+      changed_when: false
 
     - name: Extract node groups
       set_fact:
         cluster_group: "{{ stack_output.stdout | from_json }}"
   when: cluster_state == 'query'
 
+- name: Extract server ids from cluster group output
+  set_fact:
+    cluster_servers: "{{ cluster_group.output_value | json_query('[*].nodes[*].id') | flatten }}"
```
Comment (markgoddard) [inline tasks/main.yml]: json_query requires jmespath to be installed. Could you add that to the readme?

File: tasks/main.yml [hunk 2]
```diff
@@ -34,12 +34,32 @@
...
+- name: Attach interfaces to servers
+  os_server_interface:
+    auth_type: "{{ cluster_auth_type or omit }}"
+    auth: "{{ cluster_auth or omit }}"
+    cloud: "{{ cluster_cloud or omit }}"
+    state: "{{ cluster_state }}"
+    server_id: "{{ item }}"
+    interfaces: "{{ cluster_interfaces }}"
+  with_items: "{{ cluster_servers }}"
+  register: openstack_server_interfaces
```
Comment (markgoddard) [inline tasks/main.yml]: Does this need when: cluster_state in ['present', 'query'] ?

Comment (brtknr) [reply tasks/main.yml]: Yep, in query state, it doesn't add new interfaces, only gets a list of interfaces attached to a server.

File: tasks/main.yml (hunk 1 above)
Comment (brtknr) [reply tasks/main.yml]: Done

File: tasks/main.yml [hunk 3]
```diff
@@ -46,6 +47,26 @@
   when: cluster_venv != None
 
 - block:
+    - name: Extract server ids from cluster group output
+      set_fact:
+        cluster_servers: "{{ cluster_group.output_value | json_query('[*].nodes[*].id') | flatten }}"
+        cluster_interfaces: "{{ cluster_net | map(attribute='net') | list }}"
+
+    - name: Attach interfaces to servers
+      os_server_interface:
```
Comment (markgoddard) [inline tasks/main.yml]: I think this needs to run before 'Reset the python interpreter', to ensure we have access to the packages in the virtualenv.

Comment (brtknr) [reply tasks/main.yml]: Done

File: README.md [hunk 4]
```diff
@@ -12,6 +12,10 @@ The OpenStack APIs should be accessible from the target host.  OpenStack
 Newton or later is required.  Client credentials should have been set
 in the environment, or using the `clouds.yaml` format.
 
+In terms of Python `pip` packages, the requirements are:
```
Comment (markgoddard) [inline README.md]: nit: they're just Python packages

File: README.md [hunk 5]
```diff
@@ -12,6 +12,10 @@ The OpenStack APIs should be accessible from the target host.  OpenStack
...
+- ansible
+- jmespath (required by `json_query` filte)
```
Comment (markgoddard) [inline README.md]: nit: filter

**Pull request merged**